R/module_teal.R

```diff
@@ -136,6 +138,29 @@ srv_teal <- function(id, modules, teal_data_rv, filter = teal_slices()) {
       title = "SessionInfo"
     )
 
+    output$lockFile <- downloadHandler(
+      filename = function() {
+        "renv.lock"
+      },
+      content = function(file) {
+        user_lockfile <- getOption("teal.renv.lockfile", "")
+        if (!file.exists(user_lockfile)) {
```

Comment:
I think this should be more like (pseudo-code):

```
if (identical(user_lockfile, "") {
  ...
} else {
  if (file.exists(user_lockfile)) { # <- we might extend this condition for file correctness in the future if needed; this will create a space for it
    user_lockfile
  } else {
    stop("provided lock file does not exists")
  }
}
```

This is to distinguish not specified option versus not accessible / readable file path. Right now these are combined into one.

Reply:
Thanks for pointing to that! I handled this in `teal_lockfile()` in the end
d2da42b31d9ad2c52624ba76cb4ce60c59c0bd71